Sources/Diagnostics/DiagnosticsFileHandler.swift

```diff
@@ -74,20 +77,30 @@ actor DiagnosticsFileHandler: DiagnosticsFileHandlerType {
         }
 
         do {
-            try await fileHandler.removeFirstLines(diagnosticsSentCount)
+            try await self.fileHandler.removeFirstLines(diagnosticsSentCount)
         } catch {
             Logger.error("Failed to clean sent diagnostics: \(error.localizedDescription)")
         }
     }
 
     func emptyDiagnosticsFile() async {
         do {
-            try await fileHandler.emptyFile()
+            try await self.fileHandler.emptyFile()
         } catch {
             Logger.error("Failed to empty diagnostics file: \(error.localizedDescription)")
         }
     }
 
+    func isDiagnosticsFileTooBig() async -> Bool {
+        do {
+            return try await self.fileHandler.fileSizeInKB() > Self.maxFileSizeInKb
+        } catch {
+            Logger.error("Failed to check whether diagnostics file is too big: \(error.localizedDescription)")
```

Comment:
kinda surprised we haven't extracted the strings for error messages in this file like we do for all the others

Reply:
Good point! Addressed in https://github.com/RevenueCat/purchases-ios/pull/3824/commits/b452ebdf21ca1067772332f5655c47aba374c58d